pulsar-metadata/src/main/java/org/apache/pulsar/metadata/api/MetadataStore.java

```diff
@@ -112,6 +112,15 @@
      */
     CompletableFuture<Void> delete(String path, Optional<Long> expectedVersion);
 
+    /**
+     * Delete a key-value pair and all the children nodes
```

Comment:
Shall we add a specification that in case of failure the result of the operation is undetermined?
We are not using a single transaction in zk and so the deletion may be partial

Reply:
Good point, let me update the javadoc